modules/elastic-index/main.tf

```diff
@@ -0,0 +1,90 @@
+locals {
+  credentials_data    = jsondecode(data.ibm_resource_key.elastic_credentials.credentials_json).connection.https
+  elastic_auth_base64 = sensitive(base64encode("${local.credentials_data.authentication.username}:${local.credentials_data.authentication.password}"))
+  # Compose the URL without credentials to keep the latter sensitive
+  elastic_url         = "${local.credentials_data.scheme}://${local.credentials_data.hosts[0].hostname}:${local.credentials_data.hosts[0].port}"
+  elastic_ca_cert     = local.credentials_data.certificate.certificate_base64
+  elastic_instance_id = split(":", var.elastic_instance_crn)[7]
+}
+
+data "ibm_resource_key" "elastic_credentials" {
+  resource_instance_id = var.elastic_instance_crn
+  name                 = var.elastic_credentials_name
+}
+
+# Elastic index creation
+# Note: Elastic uses CA certificate for connection, and restapi does not have an option to specify that
+# Therefore the Elastic APIs are used from shell_script
+resource "shell_script" "elastic_index" {
```

Comment:
Would it be possible to use the following tf provider https://registry.terraform.io/providers/phillbaker/elasticsearch/latest/docs . Going tf native would typically be the recommended way as opposed to using bash scripts (last resort)

Reply:
@vburckhardt - I think I'd rather use https://registry.terraform.io/providers/elastic/elasticstack/latest/docs/resources/elasticsearch_index (unless it wouldn't work with IDC). The problem is that we upload data to elastic in that same module, and none of the providers do that. Eventually we probably should move the upload to the pipeline, since it's not really a "resource".
So we can make the index provisioning a native TF, but the data upload will still be a shell script.